src/main/java/org/broadinstitute/dsm/model/familymember/AddFamilyMember.java

```diff
@@ -0,0 +1,169 @@
+import org.broadinstitute.dsm.model.rgp.RgpAddFamilyMember;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.ParticipantUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AddFamilyMember {
+
+    private static final Logger logger = LoggerFactory.getLogger(AddFamilyMember.class);
+
+    protected AddFamilyMemberPayload addFamilyMemberPayload;
+    protected ParticipantData participantData;
+    protected int ddpInstanceId;
+    protected DDPInstanceDao ddpInstanceDao;
+    protected String studyGuid;
+    protected String participantId;
+
+    protected AddFamilyMember(AddFamilyMemberPayload addFamilyMemberPayload) {
+        this.addFamilyMemberPayload = Objects.requireNonNull(addFamilyMemberPayload);
+        this.participantData = new ParticipantData();
+        ddpInstanceDao = new DDPInstanceDao();
+        studyGuid = addFamilyMemberPayload.getRealm().orElseThrow();
+        participantId = addFamilyMemberPayload.getParticipantId().orElseThrow();
+        ddpInstanceId = ddpInstanceDao.getDDPInstanceIdByGuid(studyGuid);
+    }
+
+    public long add() {
+        prepareFamilyMemberData();
+        copyProbandData();
+        addDefaultOptionsValueToData();
+        long createdParticipantDataId = this.participantData.insertParticipantData(
+                new UserDao().get(addFamilyMemberPayload.getUserId().orElse(0)).flatMap(UserDto::getEmail).orElse("SYSTEM"));
+        exportDataToEs();
+        return createdParticipantDataId;
+    }
+
+    protected void prepareFamilyMemberData() {
+        FamilyMemberDetails familyMemberDetails = addFamilyMemberPayload.getData().orElseThrow();
+        String fieldTypeId =  studyGuid + ParticipantData.FIELD_TYPE;
+        participantData.setDdpParticipantId(participantId);
+        participantData.setDdpInstanceId(ddpInstanceId);
+        participantData.setFieldTypeId(fieldTypeId);
+        familyMemberDetails.setFamilyId(addFamilyMemberPayload.getOrGenerateFamilyId());
+        familyMemberDetails.setCollaboratorParticipantId(addFamilyMemberPayload.generateCollaboratorParticipantId());
+        if (FamilyMemberConstants.MEMBER_TYPE_SELF.equalsIgnoreCase(familyMemberDetails.getMemberType()))
+            familyMemberDetails.setEmail(ParticipantUtil.getParticipantEmailById(
+                    ddpInstanceDao.getEsParticipantIndexByStudyGuid(studyGuid).orElse(""),
+                    addFamilyMemberPayload.getParticipantId().orElse("")));
+        this.participantData.setData(familyMemberDetails.toMap());
+    }
+
+    protected void copyProbandData() {
+        boolean isCopyProband = addFamilyMemberPayload.getCopyProbandInfo().orElse(Boolean.FALSE);
+        if (!isCopyProband || StringUtils.isBlank(addFamilyMemberPayload.getParticipantId().orElse(""))) return;
+        Map<String, String> participantDataData = participantData.getData();
+        if (Objects.isNull(participantDataData)) throw new NoSuchElementException();
```

Comment:
I have to say, I am not a fan of "if" without "{ }"

Reply:
And in my opinion when there is only one line it is much more readable than wrapping in curly braces, it would be perfect if we would make some kind of planning session and would agree on some code writing convention on which whole team would follow.

Reply:
oh, I think we already said we want to do that… but then … maybe after RGP is all set we should do that